src/netaddress.h

```diff
@@ -16,21 +16,47 @@
 #include <string>
 #include <vector>
 
+/**
+ * A network type.
+ * @note An address may belong to more than one network, for example `10.0.0.1`
+ * belongs to both `NET_UNROUTABLE` and `NET_IPV4`.
+ * Keep these sequential starting from 0 and `NET_MAX` as the last entry.
+ * We have loops like `for (int i = 0; i < NET_MAX; i++)` that expect to iterate
+ * over all enum values and also `GetExtNetwork()` "extends" this enum by
+ * introducing standalone constants starting from `NET_MAX`.
+ */
 enum Network
 {
+    /// Addresses from these networks are not publicly routable on the global Internet.
     NET_UNROUTABLE = 0,
+
+    /// IPv4
     NET_IPV4,
+
+    /// IPv6
     NET_IPV6,
+
+    /// TORv2
     NET_ONION,
+
+    /// A set of dummy addresses that map a name to an IPv6 address.
     NET_INTERNAL,
```

Comment:
Suggested change:
/// A set of special crafted FC00/7 IPv6 addresses we use to map a string or FQDN to an IPv6 address.
/// We uses these important fake addresses also in CAddrMan to keep track of which DNS seeds were used 
/// and how we manage buckets and peers.

 With that in mind when dev  we can take special care in transit to ADDRv2 to not mess with this address realm,
because we need them later to get our new long addresses in and out from peers,dat ?

Reply:
I extended the comment, thanks!

In https://github.com/bitcoin/bitcoin/pull/19031 we drop the 6-byte `fd6b:88c0:8724` prefix from these addresses and store them in memory as `m_net=NET_INTERNAL, m_addr=...10 bytes...`. Serializing that to `addrv1` remains unchanged (we add the `fd6b:88c0:8724` prefix) and they are never serialized as `addrv2`.